sda-commons-client-jersey/src/main/java/org/sdase/commons/client/jersey/builder/FluentApiClientBuilder.java

```diff
@@ -0,0 +1,142 @@
+package org.sdase.commons.client.jersey.builder;
+
+import javax.ws.rs.client.ClientRequestFilter;
+import org.glassfish.jersey.client.ClientProperties;
+import org.sdase.commons.client.jersey.ApiHttpClientConfiguration;
+
+/**
+ * Entrypoint to the creation of API clients based on annotated interfaces.
+ *
+ * <p>This interfaces and it's inner interfaces describe the fluent API to create clients.
+ *
+ * <p>As default, the client will follow redirects so that redirect status codes are automatically
+ * resolved by the client. Automatically following redirects only works well for 303 See Other
+ * although the {@linkplain ClientProperties#FOLLOW_REDIRECTS documentation} describes 3xx
+ * redirects. It is required to keep {@linkplain
+ * io.dropwizard.client.JerseyClientConfiguration#setGzipEnabledForRequests(boolean) GZIP disabled
+ * for requests} which is the default of the {@link ApiHttpClientConfiguration}.
+ *
+ * <p>Example usages:
+ *
+ * <pre>
+ *   <code>
+ *
+ *     // backward compatible
```

Comment:
Not sure if we use this javadoc as a migration guide. Should we just include the two examples with `// api client for platform use` and `// api client for external use`?